### PR TITLE
chore: clean up and reorganize code

### PR DESCRIPTION
**Pull request by arjunr00**
Just some fixes regarding typing errors and insufficient null safety. Also reorganized a tad to better adhere to [our design guidelines](DESIGN_GUIDELINES.md).

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -16,12 +16,13 @@
...
+    "moduleResolution": "node",
     "composite": true,
     "incremental": true,
     "typeRoots": [
       "./node_modules/@types",
       "./custom_typing"
     ]
   },
-  "includes": ["**/*.ts", "**/*.tsx"]
+  "include": ["**/*.ts", "**/*.tsx", "src/**/*.json"]
```
Comment (BryanPan342) [inline tsconfig.json:27]: does adding the json fuck with the linter rules?

File: src/global.scss [hunk 2]
```diff
@@ -46,7 +46,7 @@ a:hover {
 .section {
   background-color: $section-bg;
   box-sizing: border-box;
-  min-height: calc(100vh - #{$navbar-height});
+  min-height: calc(100vh - #{$navbar-height} + 1px);
```
Comment (BryanPan342) [inline src/global.scss:49]: why +1?

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -86,7 +91,9 @@ function Navbar(props: NavbarProps): React.Component {
   }, [ scrollTop ]);
 
   const scrollToElement = (el: HTMLElement) => {
-    const navbarHeight = document.getElementById('navbar').offsetHeight;
+    const navbar = document.getElementById('navbar');
+    const navbarHeight
+      = navbar ? navbar.offsetHeight : document.body.offsetHeight - el.offsetHeight;
```
Comment (BryanPan342) [inline src/components/Navbar.tsx:96]:
```suggestion
    const navbarHeight = navbar.offsetHeight ?? document.body.offsetHeight - el.offsetHeight;
```

**Review by BryanPan342 — DISMISSED** [review]
Just a single nit!

What was the motivation to switch components to a higher-order class?

File: tsconfig.json (hunk 1 above)
Comment (arjunr00) [reply tsconfig.json:27]: Kind of, it's just to satisfy some TypeScript import rules regarding JSON

File: src/global.scss (hunk 2 above)
Comment (arjunr00) [reply src/global.scss:49]: ignore this, it was cause the border for the next section was showing up when the previous section filled the screen, but I think a better fix is switching from `border` to `outline`

**Review by BryanPan342 — APPROVED** [review]
LGTM

**Pull request merged**